Numpy vectorized efd

reinvantveer opened the pull request:
Numpy vectorized refactoring of efd function. Includes test for 3-4 times performance increase.

Thanks for sharing your code!

hbldh commented:
This is very nice! Something I should have done when converting the Matlab code and never got around to doing. I will merge this tomorrow to version 1.30, and later on fix the CI kerfuffle. Thank you very much!

reinvantveer commented:
Hi @hbldh I added a few commits to make the tests pass. I removed testing for python versions 3.3 and 2.6, which is necessary to run the tests on an image that allows testing for python 3.6 and 3.7, versions that are much more likely to be in current use. 

Adding tests on python 3.7 requires the `dist: xenial` option (see [docs](https://docs.travis-ci.com/user/languages/python/#python-37-and-higher)), but there's no support for python 2.6 there. On the flipside, current pytest has no support for python 3.3 or 2.6 anymore (see [pypi/pytest](https://pypi.org/project/pytest/#features). So I took the liberty to ditch 3.3 and 2.6. 

Hope you approve!

hbldh commented:
I appreciate the Travis fixes: I am trying to migrate from Travis to Azure Pipelines whenever I do a new release of a project now, so there is some extra work to be done before I am done.

2.6, 3.3, 3.4 is more than fine to drop.

This code is now included in the 1.3.0 release on PyPI. Thanks once again!

reinvantveer commented:
Great!
